scripts/lint_jjb.py

```diff
@@ -225,6 +231,110 @@ def check_timed_trigger_value(schedule, data):
     return return_value
 
 
+# Checks the project for the proper url protocol.
+# Returns True if job has a invalid url protocol, otherwise returns False
+#
+def invalid_project_protocol(job, file_name):
+    error_message = ("{f}/{j}: Invalid protocol - "
+                     " only https protocol is allowed."
+                     " Expected value similar to "
+                     " https://github.com/..."
+                     " but received: "
+                     .format(j=job['name'], f=file_name))
+
+    # Looking for urls - could be any one of:
+    #
+    # repo:
+    #   - rpc-maas:
+    #       repo_url: 'https://github.com/rcbops/rpc-maas'
+    # or:
+    # repo:
+    #   - rpc-ceph:
+    #       URL: "https://github.com/rcbops/rpc-ceph"
+    #       BRANCH: "master"
+    # or:
+    # repo_name:
+    #   - rpc-uam:
+    #       repo_url: "git@github.com:rcbops/rpc-uam"
+    #       branch: master
+    if 'repo' in job:
```

Comment:
This also needs to find top level repo_url keys. A JMESPath expression like `[repo_url, [repo_name, repo][*][*].*[URL, repo_url]][][][][]` could help, or you might think thats perl style ugly.

Reply:
Yeah, the Perl path is a bit difficult to follow. My latest commit has it split out a bit more and is slightly easier to add additional paths, test and to validate.